### PR TITLE
Update Blazor Call web API

### LINKED ISSUE #16609: Further distinguish client request approaches
@Ozkr16 remarks at https://github.com/aspnet/AspNetCore.Docs/pull/12477#issuecomment-576039597 ...

&gt; When reading thought the documentation I was unable to find something like this.

... referring to DR's remarks on the PR, which were addressed. Effort was made to clarify that the topic applies to Blazor WebAssembly:

* Readers looking for the approach for Blazor Server are directed to the *Make HTTP requests using IHttpClientFactory* topic.
* A couple of sections call out that the the content applies to Blazor WebAssembly.

I intend to make the opening two paragraphs distinguish the approaches even further.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8b5e2e10-ff7b-2b9f-84da-f16028ae2c53
* Version Independent ID: c7e59a08-1c60-32c2-75fd-33cb77ff7a5d
* Content: [Call a web API from ASP.NET Core Blazor](https://docs.microsoft.com/en-us/aspnet/core/blazor/call-web-api?view=aspnetcore-3.1)
* Content Source: [aspnetcore/blazor/call-web-api.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/blazor/call-web-api.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #16609

Thanks @Ozkr16! ⛄️

### CODE REVIEW

**Comment by Ozkr16** [comment]
I found the change very clear and understandable. Any reason to reopen the PR?

**Comment by guardrex** [comment]
There was a build error. You how sometimes hitting something with a 🔨 will free it? .... Sometimes when we close and re-open a PR, it clears the build error.

File: aspnetcore/blazor/hosting-models.md [hunk 1]
```diff
@@ -492,6 +492,26 @@ To configure the SignalR client in the *Pages/_Host.cshtml* file:
 </script>
 ```
 
+## Conditionally execute code based on location
```
Comment (danroth27) [inline aspnetcore/blazor/hosting-models.md]: @SteveSandersonMS @lewing is there any additional guidance that we want to provide here about when it's appropriate to do this?

Comment (danroth27) [reply aspnetcore/blazor/hosting-models.md]: @guardrex I think we should hold off on documenting this technique for platform detection. I'm not convinced the runtime is doing what we want here: https://github.com/mono/mono/issues/18627.

**Comment by guardrex** [comment]
@danroth27 *Done* :+1:

**Comment by guardrex** [comment]
@danroth27 Did you want to take a last look? :eyes:

**Pull request merged**